### PR TITLE
updated for new class structure

### PR DESCRIPTION
**Pull request by andrewjmeier**
I updated Card, Player, and Game to use the new class definition style. This allows the class to be instantiated using `var game = new Game(players, ...)` instead of doing 
```var game = Object.create(Game); 
game.players = players;
...```
I also added a bunch of tests for the chance/community chest cards to make sure they are working properly.

### CODE REVIEW

File: src/game.js [hunk 1]
```diff
@@ -1,74 +1,81 @@
...
+Game.prototype.movePlayer = function() {
+    this.rollDice(2);
+    if (this.players[this.currentPlayer].inJail) {
+        this.players[this.currentPlayer].turnsInJail += 1;
+        if (this.isDoubles(this.dice)) {
+            this.players[this.currentPlayer].releaseFromJail();
+            this.move();
+        } else if (this.players[this.currentPlayer].turnsInJail === 3) {
+            this.players[this.currentPlayer].payBail();
             this.move();
```
Comment (konnorbeard) [inline src/game.js]: could this.move() just go outside this entire if... if..else? looks like you call it in every case

**Comment by konnorbeard** [comment]
@andrewjmeier I only briefly looked over it but i like the changes, looks good

File: src/game.js (hunk 1 above)
Comment (andrewjmeier) [inline src/game.js]: There's one case where it doesn't get called when the player is in jail and they don't roll doubles and they don't have to pay bail. (the inner else that isn't written)

**Comment by qstearns** [comment]
Looks good to me too. I'm gonna go ahead and merge

**Pull request merged**